### PR TITLE
Apply hlint suggestions and minor format fixes 

### PR DESCRIPTION
**Pull request by Avi-D-coder**

### CODE REVIEW

File: src/Haskell/Ide/Engine/Plugin/HieExtras.hs [hunk 1]
```diff
@@ -82,8 +82,8 @@ getDynFlags = ms_hspp_opts . pm_mod_summary . tm_parsed_module
 
 -- ---------------------------------------------------------------------
 
-data NameMapData = NMD
-  { inverseNameMap ::  !(Map.Map Name [SrcSpan])
+newtype NameMapData = NMD
+  { inverseNameMap :: Map.Map Name [SrcSpan]
```
Comment (alanz) [inline src/Haskell/Ide/Engine/Plugin/HieExtras.hs]: You have removed a strictness annotation here. This is probably not intended.

Comment (alanz) [reply src/Haskell/Ide/Engine/Plugin/HieExtras.hs]: See also https://github.com/ndmitchell/hlint/issues/548

Comment (Avi-D-coder) [reply src/Haskell/Ide/Engine/Plugin/HieExtras.hs]: My bad. I changed it back.

Comment (Avi-D-coder) [reply src/Haskell/Ide/Engine/Plugin/HieExtras.hs]: For the time being I added annotations to ignore this use of `data` and the compatibility lambdas.

**Pull request merged**